zcash_client_backend/src/fees/fixed.rs

```diff
@@ -52,115 +52,19 @@ impl ChangeStrategy for SingleOutputChangeStrategy {
         dust_output_policy: &DustOutputPolicy,
         padding_rules: &PaddingRules,
     ) -> Result<TransactionBalance, ChangeError<Self::Error, NoteRefT>> {
-        let t_in = transparent_inputs
-            .iter()
-            .map(|t_in| t_in.coin().value)
-            .sum::<Option<_>>()
-            .ok_or(BalanceError::Overflow)?;
-        let t_out = transparent_outputs
-            .iter()
-            .map(|t_out| t_out.value())
-            .sum::<Option<_>>()
-            .ok_or(BalanceError::Overflow)?;
-        let sapling_in = sapling_inputs
-            .iter()
-            .map(|s_in| s_in.value())
-            .sum::<Option<_>>()
-            .ok_or(BalanceError::Overflow)?;
-        let sapling_out = sapling_outputs
-            .iter()
-            .map(|s_out| s_out.value())
-            .sum::<Option<_>>()
-            .ok_or(BalanceError::Overflow)?;
-
-        let fee_amount = self
-            .fee_rule
-            .fee_required(
-                params,
-                target_height,
-                transparent_inputs,
-                transparent_outputs,
-                sapling_inputs.len(),
-                padding_rules
-                    .sapling()
-                    .num_outputs(sapling_outputs.len() + 1),
-                //Orchard is not yet supported in zcash_client_backend
-                0,
-            )
-            .unwrap(); // fixed::FeeRule::fee_required is infallible.
-
-        let total_in = (t_in + sapling_in).ok_or(BalanceError::Overflow)?;
-
-        if (!transparent_inputs.is_empty() || !sapling_inputs.is_empty()) && fee_amount > total_in {
-            // For the fixed-fee selection rule, the only time we consider inputs dust is when the fee
```

Comment:
This case has been eliminated in this refactoring, because it doesn't matter: under a circumstance where the fee amount is greater than the total of the inputs, the proposed change calculation would return a negative value and thus return an insufficient funds error.